### ISSUE TITLE
Confusing/undocumented project layout and dependencies for local development

### ISSUE THREAD

**Issue by max-arnold** [yourlabs/django-cities-light#111]
There is no clear/documented way for new developers to start hacking this project. Let me pretend to be a newbie:

- This is fairly standard way for me to install the project locally (no hints in the docs though):
```
% git clone git@github.com:yourlabs/django-cities-light.git
% cd django-cities-light
% mkvirtualenv django-cities-light
% setvirtualenvproject
% pip install -e ./
```

- Ok, how to run this thing?

```
% find . -name manage.py
./test_project/manage.py
```

- Great, we have manage.py, let's try it:

```
% test_project/manage.py
Traceback (most recent call last):
  File "test_project/manage.py", line 8, in <module>
    from django.core.management import execute_from_command_line
ImportError: No module named django.core.management
```

- It looks like we need to install Django

```
% pip install django
Collecting django
  Using cached Django-1.9.4-py2.py3-none-any.whl
Installing collected packages: django
Successfully installed django-1.9.4
```

- Trying again

```
% test_project/manage.py
Traceback (most recent call last):
  File "test_project/manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/django/core/management/__init__.py", line 353, in execute_from_command_line
    utility.execute()
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/django/core/management/__init__.py", line 302, in execute
    settings.INSTALLED_APPS
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/django/conf/__init__.py", line 55, in __getattr__
    self._setup(name)
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/django/conf/__init__.py", line 43, in _setup
    self._wrapped = Settings(settings_module)
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/django/conf/__init__.py", line 99, in __init__
    mod = importlib.import_module(self.SETTINGS_MODULE)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
ImportError: No module named test_project.settings
```

- Where are the settings? Why test_project doesn't have them?

```
% find . -name 'settings*.py'
./cities_light/settings.py
./cities_light/tests/project/settings.py
./cities_light/tests/project/settings_mysql.py
./cities_light/tests/project/settings_postgres.py
./cities_light/tests/project/settings_sqlite.py
```

- Here they are! Let's try again

```
% test_project/manage.py --settings cities_light.tests.project.settings
Traceback (most recent call last):
  File "test_project/manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/django/core/management/__init__.py", line 353, in execute_from_command_line
    utility.execute()
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/django/core/management/__init__.py", line 302, in execute
    settings.INSTALLED_APPS
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/django/conf/__init__.py", line 55, in __getattr__
    self._setup(name)
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/django/conf/__init__.py", line 43, in _setup
    self._wrapped = Settings(settings_module)
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/django/conf/__init__.py", line 99, in __init__
    mod = importlib.import_module(self.SETTINGS_MODULE)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
ImportError: No module named test_project.settings
```

- WTF? Let's try to run management command

```
test_project/manage.py migrate --settings cities_light.tests.project.settings
Traceback (most recent call last):
  File "test_project/manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/django/core/management/__init__.py", line 353, in execute_from_command_line
    utility.execute()
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/django/core/management/__init__.py", line 327, in execute
    django.setup()
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/django/__init__.py", line 18, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/django/apps/registry.py", line 85, in populate
    app_config = AppConfig.create(entry)
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/django/apps/config.py", line 90, in create
    module = import_module(entry)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
ImportError: No module named dbdiff
```

- Another dependency

```
% pip install dbdiff
Collecting dbdiff
  Could not find a version that satisfies the requirement dbdiff (from versions: )
No matching distribution found for dbdiff

% pip install django-dbdiff
Collecting django-dbdiff
Collecting json-delta (from django-dbdiff)
  Using cached json_delta-2.0-py2.py3-none-any.whl
Collecting ijson (from django-dbdiff)
  Using cached ijson-2.3-py2.py3-none-any.whl
Requirement already satisfied (use --upgrade to upgrade): six in /Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages (from django-dbdiff)
Installing collected packages: json-delta, ijson, django-dbdiff
Successfully installed django-dbdiff-0.6.0 ijson-2.3 json-delta-2.0
```

- Another try

```
% test_project/manage.py migrate --settings cities_light.tests.project.settings
/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/django/template/utils.py:37: RemovedInDjango110Warning: You haven't defined a TEMPLATES setting. You must do so before upgrading to Django 1.10. Otherwise Django will be unable to load templates.
  "unable to load templates.", RemovedInDjango110Warning)

Traceback (most recent call last):
  File "test_project/manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/django/core/management/__init__.py", line 353, in execute_from_command_line
    utility.execute()
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/django/core/management/__init__.py", line 345, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/django/core/management/base.py", line 348, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/django/core/management/base.py", line 398, in execute
    self.check()
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/django/core/management/base.py", line 426, in check
    include_deployment_checks=include_deployment_checks,
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/django/core/checks/registry.py", line 75, in run_checks
    new_errors = check(app_configs=app_configs)
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/django/core/checks/urls.py", line 13, in check_url_config
    return check_resolver(resolver)
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/django/core/checks/urls.py", line 23, in check_resolver
    for pattern in resolver.url_patterns:
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/django/utils/functional.py", line 33, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/django/core/urlresolvers.py", line 417, in url_patterns
    patterns = getattr(self.urlconf_module, "urlpatterns", self.urlconf_module)
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/django/utils/functional.py", line 33, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/django/core/urlresolvers.py", line 410, in urlconf_module
    return import_module(self.urlconf_name)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
  File "/Users/user/play/bf/django-cities-light/cities_light/tests/project/urls.py", line 12, in <module>
    url(r'^autocomplete/', include('autocomplete_light.urls')),
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/django/conf/urls/__init__.py", line 52, in include
    urlconf_module = import_module(urlconf_module)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
ImportError: No module named autocomplete_light.urls
```

- More dependency hell

```
% pip install autocomplete-light
Collecting autocomplete-light
  Could not find a version that satisfies the requirement autocomplete-light (from versions: )
No matching distribution found for autocomplete-light

% pip install django-autocomplete-light
Collecting django-autocomplete-light
Installing collected packages: django-autocomplete-light
Successfully installed django-autocomplete-light-3.1.4
```

- Still trying

```
% test_project/manage.py migrate --settings cities_light.tests.project.settings
/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/django/template/utils.py:37: RemovedInDjango110Warning: You haven't defined a TEMPLATES setting. You must do so before upgrading to Django 1.10. Otherwise Django will be unable to load templates.
  "unable to load templates.", RemovedInDjango110Warning)

Traceback (most recent call last):
  File "test_project/manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/django/core/management/__init__.py", line 353, in execute_from_command_line
    utility.execute()
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/django/core/management/__init__.py", line 345, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/django/core/management/base.py", line 348, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/django/core/management/base.py", line 398, in execute
    self.check()
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/django/core/management/base.py", line 426, in check
    include_deployment_checks=include_deployment_checks,
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/django/core/checks/registry.py", line 75, in run_checks
    new_errors = check(app_configs=app_configs)
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/django/core/checks/urls.py", line 13, in check_url_config
    return check_resolver(resolver)
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/django/core/checks/urls.py", line 23, in check_resolver
    for pattern in resolver.url_patterns:
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/django/utils/functional.py", line 33, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/django/core/urlresolvers.py", line 417, in url_patterns
    patterns = getattr(self.urlconf_module, "urlpatterns", self.urlconf_module)
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/django/utils/functional.py", line 33, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/django/core/urlresolvers.py", line 410, in urlconf_module
    return import_module(self.urlconf_name)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
  File "/Users/user/play/bf/django-cities-light/cities_light/tests/project/urls.py", line 12, in <module>
    url(r'^autocomplete/', include('autocomplete_light.urls')),
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/django/conf/urls/__init__.py", line 52, in include
    urlconf_module = import_module(urlconf_module)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
ImportError: No module named autocomplete_light.urls
```

- After reading django-autocomplete-light commit history I know the magic version

```
% pip uninstall django-autocomplete-light

% pip install 'django-autocomplete-light<2'
Collecting django-autocomplete-light<2
Installing collected packages: django-autocomplete-light
Successfully installed django-autocomplete-light-1.4.14
```

- And...

```
% test_project/manage.py migrate --settings cities_light.tests.project.settings
/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/django/template/utils.py:37: RemovedInDjango110Warning: You haven't defined a TEMPLATES setting. You must do so before upgrading to Django 1.10. Otherwise Django will be unable to load templates.
  "unable to load templates.", RemovedInDjango110Warning)

Traceback (most recent call last):
  File "test_project/manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/django/core/management/__init__.py", line 353, in execute_from_command_line
    utility.execute()
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/django/core/management/__init__.py", line 345, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/django/core/management/base.py", line 348, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/django/core/management/base.py", line 398, in execute
    self.check()
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/django/core/management/base.py", line 426, in check
    include_deployment_checks=include_deployment_checks,
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/django/core/checks/registry.py", line 75, in run_checks
    new_errors = check(app_configs=app_configs)
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/django/core/checks/urls.py", line 13, in check_url_config
    return check_resolver(resolver)
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/django/core/checks/urls.py", line 23, in check_resolver
    for pattern in resolver.url_patterns:
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/django/utils/functional.py", line 33, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/django/core/urlresolvers.py", line 417, in url_patterns
    patterns = getattr(self.urlconf_module, "urlpatterns", self.urlconf_module)
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/django/utils/functional.py", line 33, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/django/core/urlresolvers.py", line 410, in urlconf_module
    return import_module(self.urlconf_name)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
  File "/Users/user/play/bf/django-cities-light/cities_light/tests/project/urls.py", line 12, in <module>
    url(r'^autocomplete/', include('autocomplete_light.urls')),
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/django/conf/urls/__init__.py", line 52, in include
    urlconf_module = import_module(urlconf_module)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/autocomplete_light/__init__.py", line 4, in <module>
    from .registry import AutocompleteRegistry, registry, register, autodiscover
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/autocomplete_light/registry.py", line 23, in <module>
    from .autocomplete import AutocompleteModelBase
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/autocomplete_light/autocomplete/__init__.py", line 6, in <module>
    from .generic import AutocompleteGeneric
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/autocomplete_light/autocomplete/generic.py", line 4, in <module>
    from autocomplete_light.generic import GenericModelChoiceField
  File "/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/autocomplete_light/generic.py", line 4, in <module>
    from django.contrib.contenttypes.generic import GenericForeignKey
ImportError: No module named generic
```

- Fuck!

```
% pip uninstall django<1.9

% pip install 'django<1.9'
Collecting django<1.9
  Using cached Django-1.8.11-py2.py3-none-any.whl
Installing collected packages: django
Successfully installed django-1.8.11
```

- First success!

```
% test_project/manage.py migrate --settings cities_light.tests.project.settings
Operations to perform:
  Synchronize unmigrated apps: staticfiles, messages, dbdiff
  Apply all migrations: sessions, admin, sites, auth, contenttypes, cities_light
Synchronizing apps without migrations:
  Creating tables...
    Running deferred SQL...
  Installing custom SQL...
Running migrations:
  Rendering model states... DONE
  Applying contenttypes.0001_initial... OK
  Applying auth.0001_initial... OK
  Applying admin.0001_initial... OK
  Applying contenttypes.0002_remove_content_type_name... OK
  Applying auth.0002_alter_permission_name_max_length... OK
  Applying auth.0003_alter_user_email_max_length... OK
  Applying auth.0004_alter_user_username_opts... OK
  Applying auth.0005_alter_user_last_login_null... OK
  Applying auth.0006_require_contenttypes_0002... OK
  Applying cities_light.0001_initial... OK
  Applying cities_light.0002_city... OK
  Applying cities_light.0003_auto_20141120_0342... OK
  Applying cities_light.0004_autoslug_update... OK
  Applying cities_light.0005_blank_phone... OK
  Applying cities_light.0006_compensate_for_0003_bytestring_bug... OK
  Applying sessions.0001_initial... OK
  Applying sites.0001_initial... OK
```

- Let's import some geonames data!

```
% test_project/manage.py cities_light --settings cities_light.tests.project.settings
Forced import of angouleme_country.txt because data do not seem to have installed successfuly yet, note that this is equivalent to --force-import-all.
Importing angouleme_country.txt
Saving France
/Users/user/.virtualenvs/django-cities-light/lib/python2.7/site-packages/unidecode/__init__.py:46: RuntimeWarning: Argument <type 'str'> is not an unicode object. Passing an encoded string will likely have unexpected results.
  _warn_if_not_unicode(string)

RAM used: 31801344 kB Time: 0:00:00 Done: 100%|##############################################################################################|
Forced import of angouleme_region.txt because data do not seem to have installed successfuly yet, note that this is equivalent to --force-import-all.
Importing angouleme_region.txt
Saving Poitou-Charentes
RAM used: 31858688 kB Time: 0:00:00 Done: 100%|##############################################################################################|
Forced import of angouleme_city.txt because data do not seem to have installed successfuly yet, note that this is equivalent to --force-import-all.
Importing angouleme_city.txt
Saving Angoulême
RAM used: 31928320 kB Time: 0:00:00 Done: 100%|##############################################################################################|
Forced import of angouleme_translations.txt because data do not seem to have installed successfuly yet, note that this is equivalent to --force-import-all.
Importing angouleme_translations.txt
RAM used: 31948800 kB Time: 0:00:00 Done: 100%|##############################################################################################|
Importing parsed translation in the database
RAM used: 32206848 kB Time: 0:00:00 Done: 100%|##############################################################################################|
```

- Why it is so verbose (Saving ....)? Why it is so quick?

There is non-default logging DEBUG setup. And it looks like settings.py contains non-default geonames urls! Let's edit settings.py and try to not commit modified version accidentally...

- Other issues:

1. Broken `static` and `templates` symlinks in `test_project` dir
2. `test_project/db.sqlite` blob
3. Outdated `test_project/README.rst`
4. `.tox`, `.cache` and `.coverage` are not in .gitignore
5. Test fixtures are placed in `cities_light/data` and `cities_light/fixtures`. Instead they probably should go to `cities_light/tests/data` and `cities_light/tests/fixtures`.
6. No clear separation between `test_project` and `cities_light/tests/project` (manage.py in one place, settings in other)
7. No requirements.txt in `test_project` with working dependencies (`django-dbdiff`, `django<1.9` and `django-autocomplete-light<2`, `py.test`, `mock` etc).

**Comment by jpic** [comment]
Thanks a heap for your detailed report !

I started a PR #111 which aims at simplifying all that.

Note that tests run quickly now, because they run on a subset on the data.
This is a lot nicer to travis and other travis users ;) Also, dbdiff is

Also note that tox is the easiest to create a venv, ie.:

tox -e py27-django18-mysql
source .tox/py27-django18-mysql/bin/activate
cd test_project
./manage.py ...

But I'm still working on fixing most of the issues in #111, feel free to
take a look at it

Hope this helps

**Comment by max-arnold** [comment]
The list of things to document for new developers:

1. Base system dependencies: sqlite/mysql/postgresql with passwordless login, recommended python versions to run the tests, maybe some binary libraries?
2. Either system-wide installed tox or fresh virtualenv with tox (I prefer to isolate as much dependencies as possible in venv)
3. How to populate venv with dependencies required for local development (this also should work to run tests without tox)
4. How to run the full test suite or specific test in specific environment using tox
5. How to run full local download/import cycle
6. Link to mailing list and contributor guide (https://github.com/yourlabs/community/tree/master/docs)

**Comment by jpic** [comment]
Are you certain that it's within DCL's documentation scope to teach how to
use standard python development tools ?

**Comment by max-arnold** [comment]
I think it should be like short and easy checklist plus necessary requirements, not a full-blown documentation. Also it is always helpful to know which routines and useful shortcuts are used by project maintainer. The goal is to show the simplest path and reduce surprises (like test failures due to enabled database password on local machine).

It would be really nice if you share your workflow and typical commands (like I shared mine). I can volunteer and try to summarize them in the README or docs section.

**Comment by jpic** [comment]
Awesome, so here's how tox can build a specific environment:

```
[env] 17/03 2016 21:41:54 jpic@lue ~/env/src/cities-light  (uptest_project_simplification)
$ tox -e py27-django18-sqlite
py27-django18-sqlite develop-inst-nodeps: /home/jpic/env/src/cities-light
py27-django18-sqlite installed: -f ~/.pip/downloads,coverage==4.0.3,Django==1.8.11,django-autoslug==1.9.3,-e git+git@github.com:yourlabs/django-cities-light.git@2f2bb24942c3d1933bfb813c3f4f8e967970ec6a#egg=django_cities_light,django-dbdiff==0.6.0,funcsigs==0.4,ijson==2.3,json-delta==2.0,mock==1.3.0,pbr==1.8.1,progressbar2==3.6.2,py==1.4.31,pytest==2.9.0,pytest-cov==2.2.1,pytest-django==2.9.1,six==1.10.0,Unidecode==0.4.19,wheel==0.26.0
py27-django18-sqlite runtests: PYTHONHASHSEED='1762545254'
py27-django18-sqlite runtests: commands[0] | py.test --cov ../cities_light --create-db --strict -r fEsxXw ../cities_light
===================================================== test session starts =====================================================
platform linux2 -- Python 2.7.11, pytest-2.9.0, py-1.4.31, pluggy-0.3.1
django settings: settings (from environment variable)
rootdir: /home/jpic/env/src/cities-light, inifile: 
plugins: cov-2.2.1, django-2.9.1
collected 6 items 

../cities_light/tests/test_form.py ....
../cities_light/tests/test_import.py .
../cities_light/tests/test_migrations.py .
-------------------------------------- coverage: platform linux2, python 2.7.11-final-0 ---------------------------------------
Name                                                                                                 Stmts   Miss  Cover
------------------------------------------------------------------------------------------------------------------------
/home/jpic/env/src/cities-light/cities_light/__init__.py                                                 4      0   100%
/home/jpic/env/src/cities-light/cities_light/abstract_models.py                                         77      7    91%
/home/jpic/env/src/cities-light/cities_light/admin.py                                                   40      9    78%
/home/jpic/env/src/cities-light/cities_light/contrib/__init__.py                                         0      0   100%
/home/jpic/env/src/cities-light/cities_light/contrib/ajax_selects_lookups.py                            21     21     0%
/home/jpic/env/src/cities-light/cities_light/contrib/autocompletes.py                                   23     23     0%
/home/jpic/env/src/cities-light/cities_light/contrib/restframework.py                                   44     44     0%
/home/jpic/env/src/cities-light/cities_light/contrib/restframework2.py                                  49     49     0%
/home/jpic/env/src/cities-light/cities_light/contrib/restframework3.py                                  51     51     0%
/home/jpic/env/src/cities-light/cities_light/exceptions.py                                               8      1    88%
/home/jpic/env/src/cities-light/cities_light/forms.py                                                   17      0   100%
/home/jpic/env/src/cities-light/cities_light/geonames.py                                                72     29    60%
/home/jpic/env/src/cities-light/cities_light/loading.py                                                 10      1    90%
/home/jpic/env/src/cities-light/cities_light/management/__init__.py                                      0      0   100%
/home/jpic/env/src/cities-light/cities_light/management/commands/__init__.py                             0      0   100%
/home/jpic/env/src/cities-light/cities_light/management/commands/cities_light.py                       310     63    80%
/home/jpic/env/src/cities-light/cities_light/migrations/0001_initial.py                                  6      0   100%
/home/jpic/env/src/cities-light/cities_light/migrations/0002_city.py                                     8      0   100%
/home/jpic/env/src/cities-light/cities_light/migrations/0003_auto_20141120_0342.py                       7      0   100%
/home/jpic/env/src/cities-light/cities_light/migrations/0004_autoslug_update.py                          6      0   100%
/home/jpic/env/src/cities-light/cities_light/migrations/0005_blank_phone.py                              5      0   100%
/home/jpic/env/src/cities-light/cities_light/migrations/0006_compensate_for_0003_bytestring_bug.py       7      0   100%
/home/jpic/env/src/cities-light/cities_light/migrations/__init__.py                                      5      3    40%
/home/jpic/env/src/cities-light/cities_light/models.py                                                  18      0   100%
/home/jpic/env/src/cities-light/cities_light/receivers.py                                               73      9    88%
/home/jpic/env/src/cities-light/cities_light/settings.py                                                76      1    99%
/home/jpic/env/src/cities-light/cities_light/signals.py                                                  9      0   100%
/home/jpic/env/src/cities-light/cities_light/tests/__init__.py                                           0      0   100%
/home/jpic/env/src/cities-light/cities_light/tests/test_form.py                                         30      0   100%
/home/jpic/env/src/cities-light/cities_light/tests/test_import.py                                       24      0   100%
/home/jpic/env/src/cities-light/cities_light/tests/test_migrations.py                                   18      0   100%
------------------------------------------------------------------------------------------------------------------------
TOTAL                                                                                                 1018    311    69%

================================================== 6 passed in 1.28 seconds ===================================================
___________________________________________________________ summary ___________________________________________________________
  py27-django18-sqlite: commands succeeded
  congratulations :)

[env] 17/03 2016 21:42:40 jpic@lue ~/env/src/cities-light  (uptest_project_simplification)
$ source .tox/py27-django18-sqlite/
bash: source: .tox/py27-django18-sqlite/: is a directory

[env] 17/03 2016 21:42:48 jpic@lue ~/env/src/cities-light  (uptest_project_simplification)
$ source .tox/py27-django18-sqlite/bin/activate

[py27-django18-sqlite] 17/03 2016 21:42:50 jpic@lue ~/env/src/cities-light  (uptest_project_simplification)
$ cd test_project/

[py27-django18-sqlite] 17/03 2016 21:42:54 jpic@lue ~/env/src/cities-light/test_project  (uptest_project_simplification)
$ ./manage.py migrate
Operations to perform:
  Synchronize unmigrated apps: staticfiles, messages
  Apply all migrations: sessions, admin, sites, auth, contenttypes, cities_light
Synchronizing apps without migrations:
  Creating tables...
    Running deferred SQL...
  Installing custom SQL...
Running migrations:
  Rendering model states... DONE
  Applying contenttypes.0001_initial... OK
  Applying auth.0001_initial... OK
  Applying admin.0001_initial... OK
  Applying contenttypes.0002_remove_content_type_name... OK
  Applying auth.0002_alter_permission_name_max_length... OK
  Applying auth.0003_alter_user_email_max_length... OK
  Applying auth.0004_alter_user_username_opts... OK
  Applying auth.0005_alter_user_last_login_null... OK
  Applying auth.0006_require_contenttypes_0002... OK
  Applying cities_light.0001_initial... OK
  Applying cities_light.0002_city... OK
  Applying cities_light.0003_auto_20141120_0342... OK
  Applying cities_light.0004_autoslug_update... OK
  Applying cities_light.0005_blank_phone... OK
  Applying cities_light.0006_compensate_for_0003_bytestring_bug... OK
  Applying sessions.0001_initial... OK
  Applying sites.0001_initial... OK

[py27-django18-sqlite] 17/03 2016 21:42:57 jpic@lue ~/env/src/cities-light/test_project  (uptest_project_simplification)
$ ./manage.py migrate
Operations to perform:
  Synchronize unmigrated apps: staticfiles, messages
  Apply all migrations: sessions, admin, sites, auth, contenttypes, cities_light
Synchronizing apps without migrations:
  Creating tables...
    Running deferred SQL...
  Installing custom SQL...
Running migrations:
  No migrations to apply.

[py27-django18-sqlite] 17/03 2016 21:42:58 jpic@lue ~/env/src/cities-light/test_project  (uptest_project_simplification)
$ ./manage.py cities_light
/home/jpic/env/src/cities-light/.tox/py27-django18-sqlite/lib/python2.7/site-packages/unidecode/__init__.py:46: RuntimeWarning: Argument <type 'str'> is not an unicode object. Passing an encoded string will likely have unexpected results.
  _warn_if_not_unicode(string)

RAM used: 32864 kB Time: 0:00:03 Done: 100%|##################################################################################|
RAM used: 33264 kB ETA:  0:00:51 Done:   9%|#######         
```

Looks like everything is working again now, I'd like to setup automatic deployment on openshift with full data import but that's another pr.

About this issue, is there anything else you'd like me to do ?

Thanks !

**Comment by max-arnold** [comment]
I had some troubles with documentation (`make html` failures) and fixed it in a separate fork (django 1.9 only). If you think these fixes are useful, you can port them too:

https://github.com/greenday2/django-cities-light/commit/c53eec572c5ef13dfb5298381703ea47aef2cf6e
https://github.com/greenday2/django-cities-light/commit/1632113183b98f1122c23032300044deec271dd7

Also I plan to move `angouleme_city.txt`, `angouleme_country.txt`, `angouleme_region.txt` and `angouleme_translations.txt` to `tests/fixtures/` but not in this PR (this will be done after merging #106 with refactored Downloader class which can fetch both local and remote files).

Thanks a lot!

**Comment by jpic** [comment]
I pushed the doc fixes to #114.

Thanks !
